py_ecc/bls/point_compression.py

```diff
@@ -56,10 +103,23 @@ def decompress_G1(z: G1Compressed) -> G1Uncompressed:
     """
     Recovers x and y coordinates from the compressed point.
     """
-    # b_flag == 1 indicates the infinity point
-    b_flag = (z % POW_2_383) // POW_2_382
-    if b_flag == 1:
+    c_flag = get_c_flag(z)
+    b_flag = get_b_flag(z)
+    a_flag = get_a_flag(z)
```

Comment:
What do you think if we can get more performance and readability by using bit operations?

```python
c_flag , b_flag, a_flag = get_flags(z)
def get_flags(z):
    a_flag = bool((z >> 380) & 1)
    b_flag = bool((z >> 381) & 1)
    c_flag = bool((z >> 382) & 1)
    return c_flag , b_flag, a_flag
```
```python
>>> import random
>>> for _ in range(10):
...     z = random.getrandbits(384)
...     print(get_flags(z))
... 
(True, False, True)
(True, True, True)
(True, True, False)
(False, True, False)
(False, False, False)
(True, True, True)
(True, True, False)
(True, True, False)
(True, False, True)
(False, True, True)
```